plugins/modules/foreman_role.py

```diff
@@ -85,9 +94,17 @@ def main():
             description=dict(),
             locations=dict(type='entity_list', flat_name='location_ids'),
             organizations=dict(type='entity_list', flat_name='organization_ids'),
+            filters=dict(type='entity_list', flat_name='filters'),
```

Comment:
you could (but don't have to) add sub-options to this definition, to make Ansible validate them, see

https://github.com/theforeman/foreman-ansible-modules/blob/b02a7bcff1d6648844717636c46db2d74dc5c0b7/plugins/modules/katello_content_view.py#L160-L163

and

https://github.com/theforeman/foreman-ansible-modules/blob/b02a7bcff1d6648844717636c46db2d74dc5c0b7/plugins/modules/katello_content_view.py#L50-L65

Reply:
I added filters spec below, which should also validate it on creation of filters, if I am not mistaken?

Reply:
Nope, not for nested params (yet) sadly.